geom.go

```diff
@@ -182,6 +182,20 @@ func NewRect(p Point, lengths []float64) (r *Rect, err error) {
 	return
 }
 
+// NewRectFromPoints constructs and returns a pointer to a Rect given a corner points.
+func NewRectFromPoints(minPoint, maxPoint Point) (r *Rect, err error) {
+	if len(minPoint) != len(maxPoint) {
+		err = &DimError{len(minPoint), len(maxPoint)}
+		return
+	}
+
+	r = new(Rect)
+	r.p = minPoint
+	r.q = maxPoint
+
+	return
```

Comment:
I regret using the named return values now, but it's eight years old at this point and not worth changing everywhere. Thank you for using the existing style, even if it's not great! Still, for the lines 192-194, I think it's clearer and less verbose to write:

```
r = &Rect{p: minPoint, q: maxPoint}
return
```